core/templates/pages/about-page/about-page.component.spec.ts

```diff
@@ -55,165 +60,175 @@ class MockI18nLanguageCodeService {
+      '/assets/images/path/to/image');
   });
 
-  it('should activate about license on init', () => {
-    windowRef.nativeWindow.location.hash = '#license';
-
+  it('should check user is not logged in and navigate to create lesson page',
+    fakeAsync(() => {
+      const UserInfoObject = {
+        is_moderator: false,
+        is_admin: false,
+        is_super_admin: false,
+        is_topic_manager: false,
+        can_create_collections: false,
+        preferred_site_language_code: null,
+        username: '',
+        email: '',
+        user_is_logged_in: null
+      };
+      spyOn(userService, 'getUserInfoAsync').and.returnValue(Promise.resolve(
+        UserInfo.createFromBackendDict(UserInfoObject))
+      );
+      component.ngOnInit();
+      flushMicrotasks();
+      expect(component.userIsLoggedIn).toBe(null);
+      spyOn(
+        siteAnalyticsServiceStub, 'registerCreateLessonButtonEvent')
+        .and.callThrough();
+      spyOn(global, 'setTimeout');
+      component.onClickCreateLessonButton();
+      expect(siteAnalyticsServiceStub.registerCreateLessonButtonEvent)
+        .toHaveBeenCalledWith();
+      expect(component.loginUrl).toBe('/_ah/login');
+    }));
+
+  it('should set component properties when ngOnInit() is called', () => {
     component.ngOnInit();
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#license');
-    expect(component.activeTabName).toBe('foundation');
+    expect(component.userIsLoggedIn).toBe(null);
+    expect(component.classroomUrl).toBe('/learn/math');
+    expect(component.loginUrl).toBe('/_ah/login');
   });
 
-  it('should activate foundation tab on init', () => {
-    windowRef.nativeWindow.location.hash = '#foundation';
-
+  it('should check if loader screen is working', () =>
+    fakeAsync(() => {
+      component.ngOnInit();
+      spyOn(loaderService, 'showLoadingScreen').and.callThrough();
+      expect(loaderService.showLoadingScreen)
+        .toHaveBeenCalledWith('Loading');
+    }));
+
+  it('should check if user is logged in or not', fakeAsync(() => {
+    const UserInfoObject = {
+      is_moderator: false,
+      is_admin: false,
+      is_super_admin: false,
+      is_topic_manager: false,
+      can_create_collections: true,
+      preferred_site_language_code: null,
+      username: 'tester',
+      email: 'test@test.com',
+      user_is_logged_in: true
+    };
+    spyOn(userService, 'getUserInfoAsync').and.returnValue(Promise.resolve(
+      UserInfo.createFromBackendDict(UserInfoObject))
+    );
     component.ngOnInit();
+    flushMicrotasks();
+    expect(component.userIsLoggedIn).toBe(true);
+  }));
 
-    expect(windowRef.nativeWindow.location.hash).toBe('#foundation');
-    expect(component.activeTabName).toBe('foundation');
+  it('should activate when Visit Classroom is clicked', function() {
+    spyOn(
+      siteAnalyticsServiceStub, 'registerClickVisitClassroomButtonEvent')
+      .and.callThrough();
+    component.onClickVisitClassroomButton();
```

Comment:
Can you please follow the above comments for the test files here and in all other places in this file?

Reply:
done

Reply:
Are we expecting onClickVisitClassroomButton to change the URL? IF yes then please assert those changes?

Reply:
can you please check if this is what you expect.